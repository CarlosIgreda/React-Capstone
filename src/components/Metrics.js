import React from 'react';
import { useSelector } from 'react-redux';

const Metrics = () => {
  const chosenCoin = useSelector((state) => state.metrics.data);

  if (!chosenCoin) {
    return (
      <div className="loading-container">
        <div className="spinner-border text-secondary" role="status">
          <span className="visually-hidden">Loading...</span>
        </div>
        <div>Loading...</div>
      </div>
    );
  }

  return (
    <div className="coin metrics">
      <div className="coin-symbol">{chosenCoin.coinSymbol}</div>
      <img src={chosenCoin.coinImage} alt={chosenCoin.coinName} />
      <div className="coin-name">{chosenCoin.coinName}</div>
      <ul className="ul">
        <li className="li">
          <div className="coin-price">
            Current Price:&nbsp;
            {chosenCoin.coinPrice}
          </div>
        </li>
        <li className="li">
          <div className="coin-ath">
            ATH:&nbsp;
            {chosenCoin.coinAth}
          </div>
        </li>
        <li className="li">
          <div className="coin-atl">
            ATL:&nbsp;
            {chosenCoin.coinAtl}
          </div>
        </li>
      </ul>
    </div>
  );
};

export default Metrics;
