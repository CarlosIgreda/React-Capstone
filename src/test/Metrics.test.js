import React from 'react';
import { render } from '@testing-library/react';
import { useSelector } from 'react-redux';
import Metrics from '../components/Metrics';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

test('renders metrics correctly', () => {
  useSelector.mockReturnValue({
    coinSymbol: 'BTC',
    coinImage: 'btc-image-url',
    coinName: 'Bitcoin',
    coinPrice: 50000,
  });

  const { container } = render(<Metrics />);

  expect(container.firstChild).toMatchSnapshot();
});
