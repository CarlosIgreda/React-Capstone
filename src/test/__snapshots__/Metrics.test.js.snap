// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders metrics correctly 1`] = `
<div
  class="coin metrics"
>
  <div
    class="coin-symbol"
  >
    BTC
  </div>
  <img
    alt="Bitcoin"
    src="btc-image-url"
  />
  <div
    class="coin-name"
  >
    Bitcoin
  </div>
  <div
    class="coin-price"
  >
    Current Price:Â 
    50000
  </div>
</div>
`;
